bc_obps/service/tests/test_operation_service_v2.py

```diff
@@ -295,3 +310,224 @@ def test_create_or_replace_statutory_declaration():
+            type="SFO",
+            naics_code_id=1,
+            activities=[1],
+            process_flow_diagram=MOCK_DATA_URL,
+            boundary_map=MOCK_DATA_URL,
+            equipment_list=MOCK_DATA_URL,
+        )
+        operation = OperationServiceV2.register_operation_information(
+            approved_user_operator.user.user_guid, None, payload
+        )
+        operation.refresh_from_db()
+        # check operation creation
+        assert Operation.objects.count() == 1
+        assert operation.created_by == approved_user_operator.user
+        assert operation.created_at is not None
+        assert operation.updated_by is not None  # the operation is created first, and then we add the purpose
+        # check purpose
+        assert operation.registration_purposes.count() == 1
+
+    @staticmethod
+    def test_register_operation_information_existing_operation():
+        approved_user_operator = baker.make_recipe('utils.approved_user_operator')
+        users_operation = baker.make_recipe(
+            'utils.operation', operator=approved_user_operator.operator, created_by=approved_user_operator.user
+        )
+        payload = OperationInformationIn(
+            registration_purpose='Potential Reporting Operation',
+            name="string",
+            type="SFO",
+            naics_code_id=1,
+            secondary_naics_code_id=2,
+            tertiary_naics_code_id=3,
+            activities=[1],
+            process_flow_diagram=MOCK_DATA_URL,
+            boundary_map=MOCK_DATA_URL,
+            equipment_list=MOCK_DATA_URL,
+        )
+        # check operation updates
+        operation = OperationServiceV2.register_operation_information(
+            approved_user_operator.user.user_guid, users_operation.id, payload
+        )
+        operation.refresh_from_db()
+        assert Operation.objects.count() == 1
+        assert operation.updated_by == approved_user_operator.user
+        assert operation.updated_at is not None
+        # check purpose
+        assert operation.registration_purposes.count() == 1
+
+
+class TestOperationServiceV2CreateOrUpdateOperation:
+    def test_create_operation_without_multiple_operators(self):
+        approved_user_operator = baker.make_recipe('utils.approved_user_operator')
+        payload = OperationInformationIn(
+            registration_purpose='Reporting Operation',
+            regulated_products=[1],
+            name="string",
+            type="SFO",
+            naics_code_id=1,
+            secondary_naics_code_id=2,
+            tertiary_naics_code_id=3,
+            activities=[1],
+            process_flow_diagram=MOCK_DATA_URL,
+            boundary_map=MOCK_DATA_URL,
+            equipment_list=MOCK_DATA_URL,
+        )
+        operation = OperationServiceV2.create_or_update_operation_v2(
+            approved_user_operator.user.user_guid, None, payload
+        )
+        # brianna why do i have to refresh? just a test thing or a problem?
```

Comment:
I presume you needed to do the refresh in order to get the expected values for the audit fields - this is consistent with what we see in other test files. Need to do the db refresh because the audit fields are only populated when the record is written to the db. I'm pretty sure the `operation` record that is returned from the create_or_update function only contains the data fields that we explicitly set (not the magic ones).